### DISCUSSION TITLE
All photos are disappeared

### DISCUSSION THREAD

**Discussion by pippo73** [photoprism/photoprism#3704] [Q&A] [answered]
PLEASE PROCEED ONLY IF YOU ARE ABSOLUTELY SURE THAT THIS IS NOT A TECHNICAL SUPPORT INCIDENT AND/OR POSSIBLY A PROBLEM WITH SOME OTHER SOFTWARE YOU ARE USING. VISIT <https://www.photoprism.app/kb/getting-support> TO LEARN MORE ABOUT OUR SUPPORT OPTIONS. THANK YOU FOR YOUR CAREFUL CONSIDERATION!

---------------------------------------------------------------------------

#### 1. What is not working as documented?

After importing a lot of photos,  and everything was working well, this morning I've noticed that all photos as diappeared.
How can I understand were are they and why they are lost?

#### 2. How can we reproduce it?

1. import photo
2. Search for them

#### 3. What behavior do you expect?

See my photos

#### 4. What could be the cause of your problem?

in the log file I've foung a long trace that the files are indicated as missing, but not why they are lost

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

Tell me what you need

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: PhotoPrism® CE Build [230719-73fa7bbe8 on 

(b) Database Type & Version: MariaDB,

(c) Operating System Types & Versions: Linux,

(d) Browser Types & Versions: Firefox, 

(e) Ad Blockers, Browser Plugins, and/or Firewall Software?

**Comment by lastzero** [comment]
It is probably a config issue, e.g. with the storage folders or the database. I'll convert this to a discussion for you! :)

**Reply by pippo73** [reply]
If it could help, here is my docker file

**Reply by pippo73** [reply]
[photo.zip](https://github.com/photoprism/photoprism/files/12524196/photo.zip)
Sorry I've noticed just now the file did not got attached

**Comment by dft601** [comment]
I have the same problem....  
I am very new here... I have a QNAP NAS and used your docker example file in container station.
I indexed 34.000 pictures... And everything was working as expected.

Today I got photovoltaic and I had to shut down the NAS... After that the database was empty,

It also happens, when I recreate the Photoprism (change yaml)  to change some settings.

**Reply by lastzero** [reply]
Yes, it's generally better to use an absolute path so you know where the files are and can make sure it's on a persistent volume. Since the absolute path will be different on every device, or at least for every vendor, we can't provide explicit examples for this right now. At a minimum, we would need a test device from each major vendor like QNAP, Synology, etc.

**Reply by dft601** [reply]
Yes sure. Each Hardware is different...  I also tried to change the path from MariaDB to outside the container... it "works" I get a file system, but then Photoprism itself does not work.

pippo73: I am also very new, but if you have not mounted the path for the originals to outside of the container, they still should exist in a docker volume.

**Reply by pippo73** [reply]
Unfortunately nothing left on the filesystem 😢

**Comment by pippo73** [comment]
If It may help, I have some files in the sidecar dir but not the originals anymore :-(

**Comment by lastzero** [comment, marked as answer]
Based on your feedback, we have updated the [Portainer Setup Guide](https://docs.photoprism.app/getting-started/portainer/#volumes) to include the explicit recommendation to use absolute paths on persistent volumes to store your files:

> It is important that all folders are located on a persistent volume on your device - ideally specified with an absolute path - to avoid potential data loss after a restart, e.g. if the default application folder managed by Portainer changes or gets lost for a reason beyond our control.

:point_right:  https://docs.photoprism.app/getting-started/portainer/#volumes
